src/me/remie/vulcan/leaguetasks/task/tasks/crafting/CraftLeatherBody.java

```diff
@@ -0,0 +1,173 @@
+package me.remie.vulcan.leaguetasks.task.tasks.crafting;
+
+import me.remie.vulcan.leaguetasks.LeagueScript;
+import me.remie.vulcan.leaguetasks.data.LeagueScriptConstants;
+import me.remie.vulcan.leaguetasks.helpers.BankManager;
+import me.remie.vulcan.leaguetasks.helpers.ItemDetails;
+import me.remie.vulcan.leaguetasks.helpers.shop.ShopHelper;
+import me.remie.vulcan.leaguetasks.helpers.shop.ShopType;
+import me.remie.vulcan.leaguetasks.task.LeagueTask;
+import me.remie.vulcan.leaguetasks.task.requirement.SkillRequirement;
+import simple.hooks.filters.SimpleSkills;
+import simple.hooks.queries.SimpleEntityQuery;
+import simple.hooks.wrappers.SimpleGroundItem;
+import simple.hooks.wrappers.SimpleItem;
+import simple.hooks.wrappers.SimpleNpc;
+import simple.robot.api.ClientContext;
+
+import java.util.Map;
+
+/**
+ * Created by Pepsiplaya on Mar 05, 2024 at 10:51 PM
+ */
+public class CraftLeatherBody extends LeagueTask {
+    BankManager bankManager = new BankManager(script);
+    private static final int LEATHER = 1741;
+    private static final int NEEDLE_ID = 1733;
+    private static final int THREAD_ID = 1734;
+    private static final ItemDetails NEEDLE = new ItemDetails("Needle", NEEDLE_ID, 5_000);
+    private static final ItemDetails THREAD = new ItemDetails("Thread", THREAD_ID, 5_000);
+    private final ShopHelper shopHelper;
+
+    public CraftLeatherBody(final LeagueScript script) {
+        super(script, "Craft a Leather Body");
+        this.shopHelper = new ShopHelper(this);
+        setSkillRequirement(new SkillRequirement(script.ctx, Map.of(SimpleSkills.Skills.CRAFTING, 14)));
+    }
+
+    @Override
+    public void run() {
+        if (hasItemInInventory(NEEDLE_ID) && hasItemInInventory(THREAD_ID) && hasItemInInventory(LEATHER)) {
+            if (!ctx.dialogue.dialogueOpen()) {
+                SimpleItem leather = ctx.inventory.populate().filter(LEATHER).next();
+                SimpleItem needle = ctx.inventory.populate().filter(NEEDLE_ID).next();
+                SimpleItem thread = ctx.inventory.populate().filter(THREAD_ID).next();
```

Comment:
I don't see the point of doing ` SimpleItem thread = ctx.inventory.populate().filter(THREAD_ID).next();`
You are not using it here. Because you already check if you have it above ?